src/linux/system.rs

```diff
@@ -228,10 +228,11 @@ impl System {
                         parts.next().map(|v| to_u64(v)).unwrap_or(0),
                         parts.next().map(|v| to_u64(v)).unwrap_or(0),
                         parts.next().map(|v| to_u64(v)).unwrap_or(0),
-                        frequency,
+                        get_cpu_frequency(i),
```

Comment:
This function is called way too often. Please store its result like it was done before in the loop:

```rust
let frequency = if first { get_cpu_frequency(i) } else { 0 };
```

And thus, remove the original line, which will be useless.

Reply:
Wouldn't this mean, that the cpu0 core frequency will not be udated?

Reply:
Isn't the loop starting at 0? Or did I miss something?

Reply:
Yes, the cores are found under `/sys/devices/system/cpu/cpu*/cpufreq/scaling_cur_freq` starting with 0. So on a quadcore the paths would be:
`/sys/devices/system/cpu/cpu0/cpufreq/scaling_cur_freq`
`/sys/devices/system/cpu/cpu1/cpufreq/scaling_cur_freq`
`/sys/devices/system/cpu/cpu2/cpufreq/scaling_cur_freq`
`/sys/devices/system/cpu/cpu3/cpufreq/scaling_cur_freq`